picocli-shell-jline3/build.gradle

```diff
@@ -13,7 +13,7 @@ targetCompatibility = 1.8
 
 dependencies {
     implementation     rootProject
```

Comment:
Shouldn't we change this to `api` too?

Reply:
Thanx for a review @remal ! I will push another commit for `picocli-spring-boot-starter` tomorrow (and hopefully all these commits will be squashed into one if/when this gets merged into master).

As for a `rootProject`... To be honest: this is the first that I see this Gradle directive used as a dependency (I presume that it references code/dependency  from root/src (that somehow didn't find its way into some separate submodule). 

:fast_forward: Going further: if we compare transitive dependencies for `picocli-shell-jline3` recent versions (4.5.2 vs. 4.6.0):
- https://search.maven.org/artifact/info.picocli/picocli-shell-jline3/4.5.2/jar
- https://search.maven.org/artifact/info.picocli/picocli-shell-jline3/4.6.0/jar

we will found that Maven scope changed from **_compile_** to **_runtime_** for these **_three_** dependencies: 
1. `info.picocli:picocli` (root project, mentioned above)
2. `org.jline:jline` (https://github.com/jline/jline3/blob/master/jline/pom.xml)
3. `org.jline:jline-console` (https://github.com/jline/jline3/blob/master/console/pom.xml)

It seems that Gradle configuration for all of those three dependencies should be changed from **_implementation_** to _**api** in order to re-introduce Maven scope **_compile_** (when those dependencies are used transitively, that is).

I will have to sleep on this :sleeping: